pkg/minishift/provisioner/buildroot_provisioner.go

```diff
@@ -83,28 +83,40 @@ func (p *BuildrootProvisioner) Provision(swarmOptions swarm.Options, authOptions
 	p.AuthOptions = authOptions
 	p.EngineOptions = engineOptions
 
-	log.Debugf("setting hostname %q", p.Driver.GetMachineName())
+	log.Info("\n   Setting hostname ... ")
 	if err := p.SetHostname(p.Driver.GetMachineName()); err != nil {
+		log.Info("FAIL")
 		return err
+	} else {
+		log.Info("OK")
 	}
 
-	p.AuthOptions = setRemoteAuthOptions(p)
-	log.Debugf("set auth options %+v", p.AuthOptions)
+	if err := makeDockerOptionsDir(p); err != nil {
+		return err
+	}
+
+	dockerCfg, err := p.GenerateDockerOptions(engine.DefaultPort)
+	if err != nil {
+		return err
+	}
 
-	log.Debugf("setting up certificates")
+	log.Info("Setting Docker configuration on the remote daemon...")
 
-	configureAuth := func() error {
-		if err := configureAuth(p); err != nil {
-			return &util.RetriableError{Err: err}
-		}
-		return nil
+	if _, err = p.SSHCommand(fmt.Sprintf("sudo mkdir -p %s && printf %%s \"%s\" | sudo tee %s", path.Dir(dockerCfg.EngineOptionsPath), dockerCfg.EngineOptions, dockerCfg.EngineOptionsPath)); err != nil {
+		return err
 	}
 
-	err := util.RetryAfter(5, configureAuth, time.Second*10)
-	if err != nil {
-		log.Debugf("Error configuring auth during provisioning %v", err)
+	if _, err = p.SSHCommand("sudo systemctl -f daemon-reload"); err != nil {
```

Comment:
So, only this seems to the piece of the puzzle which was required. 
It would be good to have a comment here why it is required.

Reply:
This is required because by default minikube iso doesn't have any service file and this is required in after we generate service file.